### ISSUE TITLE
Improve generated example

### ISSUE THREAD

**Issue by sixty-nine** [suchipi/hex-engine#32]
Hello, thank you for your interesting work. Javascript Daily lead me here.

In the example generated with the "First Time Setup", if the mouse goes out of the window, then the Box looses its physics.

Technically the fix is easy. In draggable.js:

```
    const onUp = () => {
        if (physics) {
            physics.setStatic(originalStatic);
        }
        isDragging = false;
    };

    mouse.onUp(onUp);
    useContext().canvas.addEventListener("mouseout", onUp);
```

I have tried to write some code to integrate this with the library, but whatever I try, I remain stuck because, the event listener must be attached to the canvas, or the Root component, yet it must change the state of the Box.

Something like this won't work because we attach the listener to the box.

```
  mouse.onUp(onUp);
  mouse.onLeave(onUp);
```

If I use 'mouse.onLeave()` in the Root component, what would be the right way to set the Box static?

I also tried to extend LowLevelMouse to provide more events, but then realized it might not be the right way to go. Still the event must be on the root and change the box, or more precisely the draggable...

I'd be willing to provide a concrete PR if you could provide some pointers.

**Comment by suchipi** [comment]
Hi! Thanks for looking into this!

Could you explain this bit further?

> Something like this won't work because we attach the listener to the box.

As an aside... something I have been meaning to do is make LowLevelMouse use only one mouse event listener per Canvas, by leveraging the "`StorageFor`" pattern found elsewhere in the codebase. If things consuming LowLevelMouse were all being called from the same mouse event listener (set up in the root), would it resolve the issue you're running into?

**Comment by sixty-nine** [comment]
So, this code will work.

```
  mouse.onUp(onUp);
  mouse.onLeave(onUp);
``` 
but if the mouse is moved too fast and exits the red box, then it will suddenly have its gravity back. The code says about Mouse.mouseLeave: "called whenever the mouse cursor exits the configured Entity's bounds". And apparently the entity is the box at this point (and that makes sense).

For your idea I will think about it. The listener should be in root, but somehow must tell all the root draggable children to restore their isStatic and reset their isDragging. I'm not sure it's possible.

**Comment by sixty-nine** [comment]
So I came up with something that seem to fix this.

I am not quite sure about the naming and how to test the generated site, I just copied what I have on my local.

Would you please have a look if I go the right way with your code?

**Comment by sixty-nine** [comment]
> As an aside... something I have been meaning to do is make LowLevelMouse use only one mouse event listener per Canvas, by leveraging the "StorageFor" pattern found elsewhere in the codebase. If things consuming LowLevelMouse were all being called from the same mouse event listener (set up in the root), would it resolve the issue you're running into?

I am not sure to understand everything about the StorageFor pattern :)

But from what I see in  "StorageForWindowSize" I guess you would still wrap the in/out detection with some logic as it is done in the Mouse component, and will cause the same problem.

You either need to provide access to lower level things (mouseover, mouseout), or find a nice abstraction like "IsMouseInsideViewport" or something.

**Comment by coyotte508** [comment]
I've also had trouble with the default `Draggable` implementation, when the mouse gets out of the canvas (below or above the viewport) while dragging the object & the mouse buton is released.

I'd love to have the default example updated to fix this issue.

**Comment by suchipi** [comment]
Fixed in c7be7996173c31a0efbce5e0022d2f76be0df794
